Common/Data/Slice.cs

```diff
@@ -444,6 +446,50 @@ public override bool TryGetValue(Symbol symbol, out dynamic data)
             return false;
         }
 
+        /// <summary>
+        /// Merge two slice with same Time
+        /// </summary>
+        /// <param name="inputSlice">slice instance</param>
+        public void MergeSlice(Slice inputSlice)
+        {
+            if (Time != inputSlice.Time)
+            {
+                throw new Exception($"Slice with time {Time} can't be merged with given slice with different {inputSlice.Time}");
```

Comment:
- Maybe using ArgumentException or InvalidOperationException might be more accurate 👌 
- `<TradeBar>/<QuoteBar>/...` think these aren't required since they are implicitly figured out by the compiler